### DISCUSSION TITLE
Screenshot to clipboard

### DISCUSSION THREAD

**Discussion by BPera1111** [andyholmes/gnome-shell-extension-gsconnect#1712] [Q&A]
Clipboard synchronization does not work with images, only with text. I use debian 12 and a tab s6 lite. Do you know if I have to make any special configuration for clipboard synchronization with images to work?

**Comment by ferdnyc** [comment]
The KDE Connect protocol [only supports text content](https://invent.kde.org/network/kdeconnect-android/-/blob/master/src/org/kde/kdeconnect/Plugins/ClibpoardPlugin/ClipboardListener.java?ref_type=heads#L96-99) for the clipboard, [in both directions](https://invent.kde.org/network/kdeconnect-android/-/blob/master/src/org/kde/kdeconnect/Plugins/ClibpoardPlugin/ClipboardPlugin.java?ref_type=heads#L69-72). Easiest way to transfer images is to use the "Send to device" helper in your file manager.

**Reply by ferdnyc** [reply]
(Actually, I think the larger issue is that _**Android**_ only supports text clipboards. If you've ever noticed, when you're viewing a photo in the Photos app or whatever... you have the option to **Share** the image to another application, but there's no option to **copy** the image. The only option you ever have for copying anything to the clipboard, when viewing non-textual content, is when it's remote media (a YouTube video, for example) you can copy the _url_ to the clipboard.)

**Reply by shardulvs** [reply]
In some android apps, I get an option to import images from clipboard so I guess android supports images in clipboard

**Reply by ferdnyc** [reply]
@shardulvs If you're _importing_ the image, it's probably not on the clipboard. (Even if you're doing it via the input method tools.) Selecting an image from your Gallery and inserting it doesn't use the clipboard, it just loads and inserts the image.

Android [doesn't really have the ability to place images _directly_ in the clipboard](https://developer.android.com/develop/ui/views/touch-and-input/copy-paste) — the clipboard can only carry text, Intents (for application shortcuts), or URIs.

It's apparently _possible_ to implement true image copy/paste on the clipboard using URIs, and there may be some apps which do that. But given all of the effort involved, I'm not too surprised support is spotty. Plus, the way image-copying has to be implemented, there's really nothing it offers that isn't also provided by the Share functionality.

Supporting clipboard image transfer requires both ends to implement all of the necessary handling themselves — there's no default implementation for them to use. So the source application needs code that knows how to turn copied images into content-provider URIs, and the target application needs its own code that understands how to use those URIs to load images and insert them in whatever way is appropriate. The _image itself_ is never on the clipboard; there has to be a local file that can be linked to via a URI. That URI goes on the clipboard, instead.

Images copied on a _remote_ device don't have local URIs, so supporting Android copy/paste would require a lot of new functionality in the KDE Connect app. Not only would it have to transfer **every** copied image onto the device's local storage _just in case_ it was going to be pasted on the Android side, but it would also have to become an image content provider so that it could generate its own URIs to the images. Normally KDE Connect doesn't directly handle images, it leaves that to the user's preferred Photo app. But it wouldn't have permission to import copied photos into a different app without user confirmation of each import, so they'd have to live in its own storage instead. ...And if it's going to store a cache of images and act as an image content provider, then it also has to manage that cache and expire outdated images, otherwise it'll grow indefinitely.
